frontend/src/concepts/pipelines/apiHooks/usePipelineVersionsForPipeline.ts

```diff
@@ -0,0 +1,23 @@
+import * as React from 'react';
+import { PipelineKF, PipelineVersionKF } from '~/concepts/pipelines/kfTypes';
+import { usePipelinesAPI } from '~/concepts/pipelines/context';
+import usePipelineQuery from '~/concepts/pipelines/apiHooks/usePipelineQuery';
+import { PipelineOptions } from '~/concepts/pipelines/types';
+
+const usePipelineVersionsForPipeline = (pipeline: PipelineKF, options?: PipelineOptions) => {
```

Comment:
We should accept the minimal param types for the API to function. Could you please update `usePipelineRunsForPipeline` at the same time.
```suggestion
const usePipelineVersionsForPipeline = (pipelineId: string, options?: PipelineOptions) => {
```